lib/extensions/postgres_cdc_stream/replication.ex

```diff
@@ -74,21 +77,25 @@ defmodule Extensions.PostgresCdcStream.Replication do
   end
 
   @impl true
-  def handle_data(<<?w, _header::192, msg::binary>>, state) do
+  def handle_data(data, state) when is_write(data) do
+    %Write{message: message} = parse(data)
+
     new_state =
-      msg
+      message
       |> Decoder.decode_message()
       |> process_message(state)
 
     {:noreply, new_state}
   end
 
-  # keepalive
-  def handle_data(<<?k, wal_end::64, _clock::64, reply>>, state) do
+  def handle_data(data, state) when is_keep_alive(data) do
+    %KeepAlive{reply: reply, wal_end: wal_end} = parse(data)
+    wal_end = wal_end + 1
+
     messages =
       case reply do
-        1 -> [<<?r, wal_end + 1::64, wal_end + 1::64, wal_end + 1::64, current_time()::64, 0>>]
-        0 -> []
+        :now -> standby(wal_end, wal_end, wal_end, :now)
```

Comment:
Mmmm, this won't ack the wal because it is not incrementing the `_end` values.

Reply:
it's incremented above - `wal_end = wal_end + 1`